app/controllers/region_admin/donors_controller.rb

```diff
@@ -0,0 +1,36 @@
+class RegionAdmin::DonorsController < ApplicationController
+  before_filter :authenticate_volunteer!
+  before_filter :authorize_region_admin!
+
+  def index
+    @locations = regional_donors
+    @regions   = available_regions
+  end
+
+  private
+
+  def regional_donors
+    if current_volunteer.super_admin?
+      Location.active.donors
+    else
+      Location.active.donors.regional(available_regions.map(&:id))
+    end
+  end
+
+  def available_regions
+    if current_volunteer.super_admin?
+      Region.all
+    else
+      current_volunteer.assignments.collect{ |a| a.admin ? a.region : nil }.compact
```

Comment:
Definitely doesn't have to be a part of this PR, but this could be simplified and made more efficient by:

```ruby
current_volunteer.assignments.eager_load(:region).where(admin: true).collect(&:region)
```

This `available_regions` method is also implemented as `Volunteer#admin_regions`. Did you consider using that here?

Without the `strict` option, `Volunteer#admin_regions` could even be implemented as a `has_many`:

```ruby
has_many :admin_regions,
         through: :assignments,
         source: :region,
         conditions: { assignment: { admin: true } }
```